### PR TITLE
Display insight class name (#137)

### PR DESCRIPTION
**Pull request by Jibbarth**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Fixed tickets | #137 

Display Insight FQCN for each section to help users to know wich Sniff/Insight is applied

### CODE REVIEW

File: src/Domain/Contracts/Insight.php [hunk 1]
```diff
@@ -22,4 +22,11 @@ public function hasIssue(): bool;
      * @return string
      */
     public function getTitle(): string;
+
+    /**
+     * Get the Full Qualified Class Name of Insight used
```
Comment (nunomaduro) [inline src/Domain/Contracts/Insight.php]:
```suggestion
     * Get the class name of Insight used.
```

File: src/Domain/Insights/Sniff.php [hunk 2]
```diff
@@ -80,4 +80,17 @@ public function getDetails(): array
             return $error->getFileInfo()->getRealPath() . ':' . $error->getLine() . ': ' . $error->getMessage();
         }, $this->errors);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getInsightClass(): string
+    {
+       return $this->getSniffClass($this->errors[0]);
```
Comment (nunomaduro) [inline src/Domain/Insights/Sniff.php]: Let's play defensive here, can you create an exception like `SniffClassNotFound` and throw it when there is no errors?

File: src/Domain/Insights/Sniff.php [hunk 3]
```diff
@@ -80,4 +80,17 @@ public function getDetails(): array
...
+    }
+
+    private function getSniffClass(Error $error): string
+    {
+        return explode('.', $error->getSourceClass())[0];
```
Comment (nunomaduro) [inline src/Domain/Insights/Sniff.php]: This logic can go to `getInsigthsClass` instead.

File: src/Domain/Insights/Sniff.php (hunk 2 above)
Comment (Jibbarth) [reply src/Domain/Insights/Sniff.php]: :+1:

**Pull request merged**